partiql-spi/src/main/kotlin/org/partiql/spi/fn/Routine.kt

```diff
@@ -0,0 +1,29 @@
+package org.partiql.spi.fn
+
+import org.partiql.types.PType
+
+/**
+ * A [Routine] is a PartiQL-routine callable from an expression context.
+ */
+public interface Routine {
+
+    /**
+     * The function name. Required.
+     */
+    public fun getName(): String
+
+    /**
+     * The formal argument definitions. Optional.
+     */
+    public fun getParameters(): Array<Parameter> = emptyArray<Parameter>()
+
+    /**
+     * The function return type. Required.
+     */
+    public fun getReturnType(): PType
+
+    /**
+     * TODO REMOVE ME ??
+     */
+    public fun getSpecific(): String
```

Comment:
Will this be kept for `v1` or is it just used as part of the refactoring process?

Reply:
Part of refactoring, removed in next PR